src/wt.ts

```diff
@@ -0,0 +1,357 @@
+}
+
+export const SEND_STARTED = "send:started";
+export const SEND_COMPLETED = "send:completed";
+export const QUEUE_COMPLETED = "queue:completed";
+export const QUEUE_CONTINUED = "queue:continued";
+
+export const VISITOR_TOKEN_KEY = "wt_visitor_token";
+export const PAGE_UUID_KEY = "wt_page_uuid";
+
+function retrieveFromCookie(key: string, config: Cookie.CookieAttributes = {}) {
+  let token = Cookie.get(key);
+  if (!token) {
+    token = uuid();
+    Cookie.set(key, token, config);
+  }
+  return token;
+}
+
+const retrieveFromQueryString = (search: string) => {
+  const qs = QS.parse(search, { ignoreQueryPrefix: true });
+  return qs.wvt as string | undefined;
+};
+
+const retrieveVisitorToken = (
+  config: Cookie.CookieAttributes = {},
+  search: string
+) =>
+  retrieveFromQueryString(search) ||
+  retrieveFromCookie(VISITOR_TOKEN_KEY, config);
+
+const retrievePageUUIDToken = () => retrieveFromCookie(PAGE_UUID_KEY);
+
+type WTConfig = {
+  cookies?: Cookies.CookieAttributes;
+  trackerUrl?: string;
+  trackerDomain?: string;
+  stringifyOptions?: QS.IStringifyOptions;
+  debounce?: {
+    min?: number;
+    max?: number;
+  };
+};
+
+/** @internal */
+export type WTContext = {
+  location: {
+    hostname: string;
+    href: string;
+    search: string;
+  };
+  document: {
+    referrer: string;
+  };
+  navigator: {
+    userAgent: string;
+  };
+  innerWidth: number;
+  innerHeight: number;
+  Image: typeof window.Image;
+};
+
+type WTEvent = {
+  kind?: string;
+  category?: string;
+  action?: string;
+  label?: string;
+  value?: string | number | boolean | null;
+  page_name?: string;
+  container?: string;
+  position?: string | number;
+  object_type?: string;
+  object_name?: string;
+  metadata?: Record<string, any>;
+  url: string;
+  referrer: string | undefined;
+  page_uuid: string | null;
+  ts: number;
+};
+
+export type WTEventParams = {
+  kind?: string;
+  category?: string;
+  action?: string;
+  label?: string;
+  value?: string | number | boolean | null;
+  pageName?: string;
+  container?: string;
+  position?: string | number;
+  objectType?: string;
+  objectName?: string;
+  [metadataKey: string]: any;
+};
+
+export type WTPayload = {
+  events: WTEvent[];
+  dimensions: {
+    width: number;
+    height: number;
+  };
+  agent: string;
+  rts: number;
+  wvt: string | undefined;
+};
+
+export class WT {
+  public loading = false;
+
+  private emitter = new EventEmitter();
+  private wtConfig: WTConfig = {
+    cookies: { expires: EXPIRES_IN_DAYS },
+  };
+  private paramDefaults = {};
+  private eventQueue: WTEvent[] = [];
+  private pageUuid: string | null = null;
+  private context: WTContext;
+  private loaderImage: HTMLImageElement | undefined;
+  private processEventsDebounced!: ReturnType<typeof debounce>;
+
+  constructor(context: WTContext) {
+    this.context = context;
+    this.updateProcessEventsDebounced();
+  }
+
+  public initialize(payload: WTConfig | ((config: WTConfig) => WTConfig)) {
```

Comment:
I know this was like this before, but can we perhaps not call this argument payload? same goes for `track()`, etc.